azure/arm/main.bicep

```diff
@@ -0,0 +1,411 @@
+  location: resourceGroup().location
+  tags: tags
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${managedIdentity.id}': {}
+    }
+  }
+  sku: {
+    name: instanceSize
+    capacity: instanceCount
+  }
+  properties: {
+    orchestrationMode: 'Uniform'
+    upgradePolicy: { mode: 'Automatic' }
+    virtualMachineProfile: {
+      osProfile: {
+        computerNamePrefix: 'agentless-scanning-'
+        adminUsername: adminUsername
+        customData: base64(customData)
+        linuxConfiguration: {
+          disablePasswordAuthentication: true
+          ssh: {
+            publicKeys: [
+              {
+                path: sshAuthorizedKeysFile
+                keyData: sshPublicKey ?? sshMockPublicKey
+              }
+            ]
+          }
+        }
+      }
+      storageProfile: {
+        osDisk: {
+          osType: 'Linux'
+          createOption: 'FromImage'
+          caching: 'ReadWrite'
+          managedDisk: {
+            storageAccountType: 'StandardSSD_LRS'
+          }
+          diskSizeGB: 30
+        }
+        imageReference: {
+          publisher: 'canonical'
+          offer: 'ubuntu-24_04-lts'
+          sku: 'server-arm64'
+          version: 'latest'
+        }
+      }
+      networkProfile: {
+        networkInterfaceConfigurations: [
+          {
+            name: 'nic'
+            properties: {
+              primary: true
+              enableAcceleratedNetworking: false
+              //disableTcpStateTracking: false
+              //enableIPForwarding: false
+              ipConfigurations: [
+                {
+                  name: 'ipconfig'
+                  properties: {
+                    privateIPAddressVersion: 'IPv4'
+                    subnet: {
+                      id: '${virtualNetwork.id}/subnets/default'
+                    }
+                  }
+                }
+              ]
+            }
+          }
+        ]
+      }
+      diagnosticsProfile: {
+        bootDiagnostics: {
+          enabled: true
+        }
+      }
+      extensionProfile: {
+        extensions: [
+          {
+            name: 'HealthExtension'
+            properties: {
+              publisher: 'Microsoft.ManagedServices'
+              type: 'ApplicationHealthLinux'
+              typeHandlerVersion: '2.0'
+              settings: {
+                protocol: 'http'
+                port: 6253
+                requestPath: '/health'
+                intervalInSeconds: 10
+                numberOfProbes: 3
+                gracePeriod: 1200
+              }
+            }
+          }
+        ]
+      }
+    }
+    automaticRepairsPolicy: {
+      enabled: true
+      gracePeriod: 'PT10M'
+    }
+  }
+}
+
+// Generate a random number from a GUID by removing the letters/hyphens and taking the first 9 digits (to avoid overflow)
+var restartMinute = int(substring(join(split(guid(vmss.id), ['a', 'b', 'c', 'd', 'e', 'f', '-']), ''), 0, 9)) % (24 * 60)
```

Comment:
not sure if the `vmss.id` will be fully random, but if you need more randomness you can use `guid(deployment().name)`

as for the way youre generating it, i dont really see a problem with the way youre doing it
```bicep
int('0x${join(split(guid(deployment().name), '-'), '')}') % 1000000000
```
should achieve the same goal but im not convinced thats better lol

Reply:
I don't really need it to be random. The goal is just to not have all scanner deployments (across all customers) restart at the exact same time. Since the VMSS ID includes the subscription ID and resource group name, the GUID will be unique enough.

The `0x` trick is nice! We still need to take a substring to avoid int overflow, though (it's not very well documented but `int` seems to be 32 bits signed).